device/core/src/internal_client.ts

```diff
@@ -162,15 +162,17 @@ export abstract class InternalClient extends EventEmitter {
    * on the Azure IoT Hub or Azure IoT Edge hub instance.
    * This method is only intended for use with Azure IoT Plug and Play.
    *
-   * @param {JSONValue} payload       A JSON-serializable object containing the telemetry to send.
-   * @param {string}    componentName The component that corresponds with the telemetry. If not specified,
-   *                                  the telemetry will be sent to the default component.
+   * @param {JSONSerializableValue}             payload               A JSON-serializable object containing the telemetry to send.
+   * @param {string}                            componentName         The component that corresponds with the telemetry. If not specified,
+   *                                                                  the telemetry will be sent to the default component.
+   * @param {Callback<results.MessageEnqueued>} sendTelemetryCallback A callback used to get notified of the success or failure of the operation.
+   *                                                                  If no callback is specified, a promise is returned instead.
    */
-  sendTelemetry(payload: JSONValue): Promise<results.MessageEnqueued>;
-  sendTelemetry(payload: JSONValue, componentName: string): Promise<results.MessageEnqueued>;
-  sendTelemetry(payload: JSONValue, sendTelemetryCallback: Callback<results.MessageEnqueued>): void;
-  sendTelemetry(payload: JSONValue, componentName: string, sendTelemetryCallback: Callback<results.MessageEnqueued>): void;
-  sendTelemetry(payload: JSONValue, callbackOrComponent?: Callback<results.MessageEnqueued> | string, sendTelemetryCallback?: Callback<results.MessageEnqueued>): Promise<results.MessageEnqueued> | void {
+  sendTelemetry(payload: JSONSerializableValue): Promise<results.MessageEnqueued>;
+  sendTelemetry(payload: JSONSerializableValue, componentName: string): Promise<results.MessageEnqueued>;
+  sendTelemetry(payload: JSONSerializableValue, sendTelemetryCallback: Callback<results.MessageEnqueued>): void;
+  sendTelemetry(payload: JSONSerializableValue, componentName: string, sendTelemetryCallback: Callback<results.MessageEnqueued>): void;
+  sendTelemetry(payload: JSONSerializableValue, callbackOrComponent?: Callback<results.MessageEnqueued> | string, sendTelemetryCallback?: Callback<results.MessageEnqueued>): Promise<results.MessageEnqueued> | void {
     let componentName: string;
```

Comment:
This is really confusing.  You have a parameter named componentName and then you have a local named componentName?!!  It would seem as though the local obscures the parameter via scoping rules.

Reply:
the naming for sure could be better, but it looks like because the previous definitions are overloads, there is not really a parameter named componentName. The user will see componentName as an option but the code itself operates on callbackOrComponent...

Reply:
Discussed over a call, but basically the answer is that `componentName` is not actually initially declared in the function scope.  Otherwise, the `let componentName: string` line would cause a compiler error. The initial variables are `payload`, `callbackOrComponent`, and `sendTelemetryCallback`.

Reply:
@YoDaMa , do you have any suggestions on alternate parameter names? Maybe something like `_0`, `_1`, `_2` or `first`, `second`, `third`?

Reply:
I wouldn't change the names.